.pre-commit-config.yaml

```diff
@@ -30,10 +30,3 @@ repos:
         examples/external-lib-interfacing/external-lib-interfacing.cpp|
         core/base/workspace_aliases.hpp
       )$
-  - id: update-ginkgo-header
```

Comment:
if we only aim for avoid pre-commit issue, we can move this to early stage such that they are checked after generation

Reply:
To extend: I think as a general rule, different pre-commit steps shouldn't undo each other's changes.

Reply:
no, it will not. unless pre-commit -> build -> pre-commit -> build.
It will not undo anything if ginkgo.hpp.in is updated there too.

Reply:
I just checked, it's even worse than that: Regardless of order, if one hook undoes the change by another hook, both of them are marked as failed and prevent the changes from being committed
```
update ginkgo header.....................................................Failed
- hook id: update-ginkgo-header
- files were modified by this hook
reuse-annotate...........................................................Failed
- hook id: reuse-annotate
- files were modified by this hook

Successfully changed header of include/ginkgo/ginkgo.hpp
Successfully changed header of include/ginkgo/core/test.hpp
```